internal/newusers/cache/getbroker.go

```diff
@@ -17,8 +19,7 @@ func (c *Cache) BrokerForUser(username string) (brokerID string, err error) {
 	err = c.db.View(func(tx *bbolt.Tx) error {
 		bucket, err := getBucket(tx, userToBrokerBucketName)
 		if err != nil {
-			c.requestClearDatabase()
-			return err
+			return errors.Join(ErrNeedsClearing, err)
```

Comment:
excellent, this is better than what I would have done naturally (I would have basically wrapped the error myself), but you correct, `errors.Join()` here is exactly the correct use case for this!